<h1> 2. $ARIMA(p,d,q)$ </h1>

<h2> 2.1 Preprocesamiento de datos y packages </h2>

In [2]:
library(ggplot2)
library(xts)
library(forecast)
library(dplyr)
library(tseries)
library(zoo)
library(quantmod)
library(TTR)
setwd("C:\\Users\\81799\\OneDrive\\Documentos\\ESFM_CLASES\\Servicio Social ARTF\\Machine Learning\\section1\\ARTF_2017_2020\\ABRIL_ARTF")
Ton_Netas_P <- read.csv("Toneladas_Netas__Productos_2014_2022.csv")#Leer DF
Ton_Netas_P <- Ton_Netas_P[-99,]#Elimino la última fila
Ton_Netas_P$X <- as.yearmon(as.factor(Ton_Netas_P$X), format ="%m/%Y")
T_N_Pro  <- Ton_Netas_P # Contiene las fechas
Ton_Netas_P <- xts(Ton_Netas_P[,c(2:6)], order.by = Ton_Netas_P$X )
T_N_P <-  data.frame(Ton_Netas_P) #Ordeno por fechas y la trae como indice
Maiz_ts <- ts(T_N_P[,1], frequency = 12, star = c(2014,1))


Dada la prueba de estacionariedad, podemos confirmar que la serie *Maiz_ ts* se puede considerar como un **proceso estacionario**, y que, a su vez este se puede modelar a partir de un proceso
$AR(p)$ y $MA(q)$; es decir, un **proceso $ARMA(p,q)$ estacionario**.
Dentro de la metodología estándar de estimación de modelos ARMA, el proceso se puede realizar a partir de candidatos posibles que puedan modelar adecuadamente el *comportamiento estocástico de la
serie*. Por lo cual propongo 5 candidatos a modelos $ARIMA(p,d,q)$, los cuales son:
* $ARIMA(1,0,1)$
* $ARIMA(1,0,0)$
* $ARIMA(2,0,1)$
* $ARIMA(1,0,2)$
* $ARIMA(0,0,1)$

El objetivo de esto será comparar y ver cual de estos candidatos tiene el menor **criterio de información**.
El código que ocuparemos para estimar los procesos ARIMA seleccionados sobre la serie es el siguiente:

In [3]:
ModA_ARIMA =arima(Maiz_ts , order = c(1,0,1))
ModB_ARIMA =arima(Maiz_ts , order = c(1,0,0))
ModC_ARIMA =arima(Maiz_ts , order = c(2,0,1))
ModD_ARIMA =arima(Maiz_ts , order = c(1,0,2))
ModE_ARIMA =arima(Maiz_ts , order = c(0,0,1))

<h2> Criterios de Información </h2>

El *Criterio de Información de Akaike(AIC)* son medidas de la calidad relativa de un modelo que representan el ajuste y el número de términs en el modelo. Este criterio se utiliza para comparar diferentes modelos. Sin embargo, el modelo con el valor más pequeño para un conjunto de predictores NO necesariamente ajusta los datos adecuadamente. Utilice tambié pruebas y gráficas de residuos para evaluar que tan bien se ajusta el modelo a los datos.

El código que ocuparemos para comparar los *criterios de información* de los modelos propuestos es:

In [5]:
ModA_ARIMA$aic
ModB_ARIMA$aic
ModC_ARIMA$aic
ModD_ARIMA$aic
ModE_ARIMA$aic


[1] 2650.104

[1] 2648.111

[1] 2651.426

[1] 2651.483

[1] 2660.408

Podemos observar que al hacer la comparación de modelos utilizando el **Criterio de Información
de Akaike**, el modelo que cuenta con el menor valor es el modelo $ARIMA(1,0,0)$, por lo cual ese es el
modelo que ocuparemos para realizar las pruebas sobre los supuestos y, en el caso de que los supuestos
se cumplan, el modelo que podremos ocupar para realizar el pronóstico de la serie.
Podemos ocupar la función *summary()* sobre el modelo seleccionado para observar las estimaciones de
los parámetros del modelo, utilizando el siguiente código:

In [6]:
summary(ModB_ARIMA)


Call:
arima(x = Maiz_ts, order = c(1, 0, 0))

Coefficients:
         ar1   intercept
      0.6152  1289609.65
s.e.  0.0796    44649.69

sigma^2 estimated as 2.979e+10:  log likelihood = -1321.06,  aic = 2648.11

Training set error measures:
                   ME     RMSE      MAE       MPE     MAPE      MASE       ACF1
Training set 2136.598 172610.3 139227.9 -1.721672 11.07217 0.8852303 0.00743361


Con lo anterior, calculamos los parametros a través de *mínimos cuadrados*.Con la ayuda de R obtuvimos la estimación de esos parametros así que tenemos que el modelo matematicamente de
$ARIMA(1,0,0)=AR(1)$ esta dado de la siguiente manera: <br> $Y_t=\Phi _0 +\Phi _1Y_{t-1}+\varepsilon _t$  <br> $Y_t=1289609.65+0.6152Y_{t-1}+\varepsilon _t$ 

<h3> Conclusión </h3>
Observamos que al momento en python nos dió el mismo modelo $ARMA(1,0)$, por lo cual concluimos que este es el modelo más adecuado.